queries.sql

```diff
@@ -8,3 +8,67 @@ SELECT name, escape_attempts from animals WHERE weight_kg > 10.5;
 SELECT * from animals WHERE neutered = true;
 SELECT * from animals WHERE NOT name = 'Gabumon';
 SELECT * from animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+-- update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+SELECT * FROM animals;
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+SELECT * FROM animals;
+
+-- delete all records in the animals table, then roll back the transaction.
+-- delete
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+-- ROLLBACK
+ROLLBACK;
+SELECT * FROM animals;
+
+-- Delete all animals born after Jan 1st, 2022
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '01-01-2022';
+SELECT * FROM animals;
+
+-- Create a save point for the transaction
+BEGIN;
+SAVEPOINT savePointOne;
+
+-- Update all animals' weight to be their weight multiplied by -1
+UPDATE animals SET weight_kg = weight_kg * -1;
+SELECT * FROM animals;
+
+-- Rolling back to the SAVEPOINT
+ROLLBACK To savePointOne;
+SELECT * FROM animals;
+
+-- Update all animals' weights that are negative to be their weight multiplied by -1
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+-- How many animals are there?
+SELECT COUNT(*) FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT MAX(escape_attempts) FROM animals WHERE neutered = true OR false;
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT MIN(weight_kg), MAX(weight_kg) FROM animals;
+
```

Comment:
- [ ] Please give maximum and minimum weight animal from each type (pokemon and digimon). You can use group by to get maximum and minimum weight from both types. :+1:

Reply:
* As stated by the previous reviewer your query needs to select ` the minimum and maximum weight of each type of animal?` which according to your transaction query below you have grouped your animals into two types `pokemon and digimon` kindly write your queries to target these two types of animals selecting their min weight and max weight grouped by species. HINT 💭 Your query can look like this `
SELECT TYPE OF ANIMAL, MIN(weight_kg), MAX(weight_kg) FROM animals
   GROUP BY TYPE OF ANIMAL;
`
![image](https://user-images.githubusercontent.com/52497246/191257056-5db27451-5dfd-40a4-abf3-760279769a45.png)

Reply:
done